File: Portfolio/Portfolio.Blazor/Data/Entities/SiteSettingsTranslation.cs
using Portfolio.Blazor.Data;

namespace Portfolio.Blazor.Data.Entities;

public sealed class SiteSettingsTranslation
{
    public int Id { get; set; }
    public int SiteSettingsId { get; set; }
    public string Locale { get; set; } = string.Empty;
    public string? CopyrightTemplate { get; set; }
    public string? MaintenanceEyebrow { get; set; }
    public string? MaintenanceTitle { get; set; }
    public string? MaintenanceDescription { get; set; }

    [JsonOrEmpty]
    public string? Seo { get; set; }

    public DateTime? CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public SiteSettings? SiteSettings { get; set; }
}

File: laravel/app/Models/SiteSettingsTranslation.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class SiteSettingsTranslation extends Model
{
    use HasFactory;

    protected $fillable = ['site_settings_id', 'locale', 'copyright_template', 'maintenance_eyebrow', 'maintenance_title', 'maintenance_description', 'seo'];

    protected $casts = ['seo' => 'array'];

    /**
     * @return BelongsTo<SiteSettings, $this>
     */
    public function siteSettings(): BelongsTo
    {
        return $this->belongsTo(SiteSettings::class);
    }
}
